include/openthread/link.h

```diff
@@ -35,6 +35,7 @@
 #ifndef OPENTHREAD_LINK_H_
 #define OPENTHREAD_LINK_H_
 
+#include <config/mac.h>
```

Comment:
Regarding the `#include <config/mac.h>`, in general we cannot add a config header in a public header file.

I understand it's added so that we can get `OPENTHREAD_CONFIG_MAC_DEFAULT_MAX_FRAME_RETRIES_DIRECT` and  `OPENTHREAD_CONFIG_MAC_DEFAULT_MAX_FRAME_RETRIES_INDIRECT` definitions  to use for the counter array but the current model won't work.

The `config/mac.h` contains only a default definition for the config options. User of OT can choose to override these default values from build system definition or through specifying a project specific config header file (`OPENTHREAD_PROJECT_CORE_CONFIG_FILE`).

In general, we do not include any config header file from the public OT header files (headers in `include/openthread` folder). In the case where user of OT provides its own project specific config header file, we technically don't own the header file and therefore cannot expose it from OT's public header files (note that other libs can include OT public headers to use OT APIs and they may have no access to the the user given project config header).

The approach/solution we have used so far is to keep the definitions in public OT header files generic:
- In cases where the API itself is tied to certain config feature, we keep the declaration of API and document the required feature in the API description (but then don't provide the implementation in the core). Example of this are [openthread/ip6.h slaac APIs](https://github.com/openthread/openthread/blob/bf4dae0c18dc84bb87091897d79d9b3d678fda66/include/openthread/ip6.h#L518-L575).
- For data members that are tied to some feature, we again include them and document that their use requires the conifg feature. Examples are "time sync" fields in [`otThreadLinkInfo`](https://github.com/openthread/openthread/blob/bf4dae0c18dc84bb87091897d79d9b3d678fda66/include/openthread/link.h#L56-L71), or `mFrameErrorRate` or `mMessageErrorRate` fields in [`otNeighborInfo`](https://github.com/openthread/openthread/blob/bf4dae0c18dc84bb87091897d79d9b3d678fda66/include/openthread/thread.h#L85-L109) definition.